x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -882,11 +973,38 @@ public static Authentication newApiKeyAuthentication(AuthenticationResult<User>
         return authentication;
     }
 
+    public Authentication toRemoteAccess(RemoteAccessAuthentication remoteAccessAuthentication) {
+        assert isApiKey() : "can only convert API key authentication to remote access";
+        final Map<String, Object> metadata = new HashMap<>(getAuthenticatingSubject().getMetadata());
+        final Authentication.RealmRef authenticatedBy = newRemoteAccessRealmRef(getAuthenticatingSubject().getRealm().getNodeName());
+        final User userFromRemoteCluster = remoteAccessAuthentication.getAuthentication().getEffectiveSubject().getUser();
+        assert userFromRemoteCluster.enabled() : "the user received from a remote cluster must be enabled";
+        final User userWithoutRoles = new User(
+            userFromRemoteCluster.principal(),
+            EMPTY_ARRAY,
+            userFromRemoteCluster.fullName(),
+            userFromRemoteCluster.email(),
+            userFromRemoteCluster.metadata(),
+            userFromRemoteCluster.enabled()
+        );
+        final Authentication authentication = new Authentication(
+            new Subject(
+                userWithoutRoles,
+                authenticatedBy,
+                Version.CURRENT,
+                remoteAccessAuthentication.copyWithRemoteAccessEntries(metadata)
+            ),
+            getAuthenticationType()
+        );
+        assert false == authentication.isAssignedToDomain();
+        return authentication;
+    }
+
     private static RealmRef maybeRewriteRealmRef(Version streamVersion, RealmRef realmRef) {
         if (realmRef != null && realmRef.getDomain() != null && streamVersion.before(VERSION_REALM_DOMAINS)) {
             logger.info("Rewriting realm [" + realmRef + "] without domain");
             // security domain erasure
-            new RealmRef(realmRef.getName(), realmRef.getType(), realmRef.getNodeName(), null);
+            return new RealmRef(realmRef.getName(), realmRef.getType(), realmRef.getNodeName(), null);
```

Comment:
Fixing this in a separate PR: https://github.com/elastic/elasticsearch/pull/93276 -- will merge 93276 first.